### DISCUSSION TITLE
How can use the mongodb session to store credential provider

### DISCUSSION THREAD

**Discussion by tufail** [nextauthjs/next-auth#1569] [Help]
Hi I am using mongodb to login with the credential provider. But after login success the Session is not creating in mongodb collection. Please help me with any example available .

here is what my code is for [...nextauth].js

 ```js
credentials: {
          email: {
            label: 'Email',
            type: 'email',
            placeholder: 'yourname@domain.com',
          },
          password: { label: 'Password', type: 'password' },
        },
        authorize: async (credentials) => { 
      ------
     return userData;
}
jwt: async (token, user, isNewUser) => {
      if (user) {
        token.uid = user.id;
      }
      return Promise.resolve(token);
    },
    session: {
      jwt: true,
      maxAge: 30 * 24 * 60 * 60, // 30 days
    },
    callbacks: {
      signIn: async (user) => {
        if (user?.id) return Promise.resolve(true);
      },
      redirect: async (url, baseUrl) => {
        return Promise.resolve('http://localhost:3000/');
      },
      session: async (session, user, sessionToken) => {
        session.foo = 'bar';
        return Promise.resolve(session);
      },
    },
    adapter: Adapters.TypeORM.Adapter(process.env.MONGODB_URL, {
      models: {
        Users: UserModel,
        Accounts: AccountModel,
        Sessions: SessionModel,
      },
    }),
    pages: {
      signIn: '/login',
      signout: '/',
    }, 
```

**Comment by StefanSelfTaught** [comment]
You are using the JSON Web Tokens as session tokens.

```js
session: {
   jwt: true,
},

**Reply by tufail** [reply]
Thank you @StefanSelfTaught . But when I am trying with `{ jwt: false}` ,getting this error 
 ``` diff
-https://next-auth.js.org/errors#callback_credentials_jwt_error Signin in with credentials is only supported if JSON Web Tokens are enabled
```

**Reply by StefanSelfTaught** [reply]
I don't think you can for now. As the docs says:

> It comes with the constraint that users authenticated in this manner are not persisted in the database, and consequently that the Credentials provider can only be used if JSON Web Tokens are enabled for sessions.

**Comment by spahrson** [comment]
So there is no way to have a custom credentials provider in combination with a database adapter in v4? can somebody explain why?

**Reply by balazsorban44** [reply]
no, at least not  built-in. check the docs for an explanation https://next-auth.js.org/providers/credentials

it's not specific for v4, it has always been the case. we advocate against rolling your own solution with passwords.
